### PR TITLE
rust: use `TARGET_LINKER` as linker

### PR DESCRIPTION
**Pull request by hingbong**
Maintainer: me / @hingbong 
Compile tested: aarch64, xiaomi ax3600, OpenWrt trunk
Run tested: aarch64, xiaomi ax3600, OpenWrt trunk

Description:

### CODE REVIEW

**Comment by hingbong** [comment]
> Using: -fuse-ld=$(TARGET_LINKER) should work without any ifdef'ery

Ok

**Comment by 1715173329** [comment]
Cc maintainer @lu-zero

File: lang/rust/rust-values.mk [hunk 1]
```diff
@@ -80,6 +80,8 @@ ifneq ($(CONFIG_RUST_SCCACHE),)
 	SCCACHE_DIR=$(RUST_SCCACHE_DIR)
 endif
 
+CARGO_RUSTFLAGS+=-Clink-arg=-fuse-ld=$(TARGET_LINKER)
```
Comment (1715173329) [inline lang/rust/rust-values.mk:83]: Please move it below `CARGO_PKG_PROFILE:=$(if $(CONFIG_DEBUG),dev,release)`, i.e. at last.

**Comment by 1715173329** [comment]
Please format your authorship and SoB.

FYI: https://openwrt.org/submitting-patches#submission_guidelines
> all commits must contain Signed-off-by: My Name \<my@email.address\> where you write your real name and real email addres

**Comment by hingbong** [comment]
> Please format your authorship and SoB.
> 
> FYI: https://openwrt.org/submitting-patches#submission_guidelines
> 
> > all commits must contain Signed-off-by: My Name <[my@email.address](mailto:my@email.address)> where you write your real name and real email addres

Ok

**Comment by 1715173329** [comment]
Well please check CI output.

**Comment by hingbong** [comment]
> Well please check CI output.

commit info changed

**Comment by lu-zero** [comment]
if it is fine for @1715173329, it is fine for me.

**Pull request merged**